### PR TITLE
STORM-1244: port backtype.storm.command.upload-credentials to java

### PR DESCRIPTION
**Pull request by abhishekagarwal87**

### CODE REVIEW

File: storm-core/src/jvm/org/apache/storm/command/UploadCredentials.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+                credentialsMap.put(rawCredentials.get(i), rawCredentials.get(i + 1));
+            }
+        }
+        if (null != credentialFile) {
+            Properties credentialProps = new Properties();
+            credentialProps.load(new FileReader(credentialFile));
+            credentialsMap.putAll(credentialProps);
+        }
+        StormSubmitter.pushCredentials(topologyName, new HashMap(), credentialsMap);
+        LOG.info("Uploaded new creds to topology: " + topologyName);
```
Comment (hustfxj) [inline storm-core/src/jvm/org/apache/storm/command/UploadCredentials.java]: LOG.info("Uploaded new creds to topology: {} ",  topologyName);

File: storm-core/src/jvm/org/apache/storm/command/UploadCredentials.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    private static final Logger LOG = LoggerFactory.getLogger(UploadCredentials.class);
+
+    public static void main(String[] args) throws Exception {
+        Map<String, Object> cl = CLI.opt("f", "file", null)
+            .arg("topologyName", CLI.FIRST_WINS)
+            .arg("rawCredentials", CLI.INTO_LIST)
+            .parse(args);
+
+        String credentialFile = (String) cl.get("f");
+        java.util.List<String> rawCredentials = (java.util.List<String>) cl.get("rawCredentials");
```
Comment (unsleepy22) [inline storm-core/src/jvm/org/apache/storm/command/UploadCredentials.java]: you can import java.util.List

File: storm-core/src/jvm/org/apache/storm/command/UploadCredentials.java (hunk 1 above)
Comment (abhishekagarwal87) [inline storm-core/src/jvm/org/apache/storm/command/UploadCredentials.java]: Done. Thanks @hustfxj

File: storm-core/src/jvm/org/apache/storm/command/UploadCredentials.java (hunk 2 above)
Comment (abhishekagarwal87) [inline storm-core/src/jvm/org/apache/storm/command/UploadCredentials.java]: There is a class List in the same package which conflicts with java.util.List

Comment (revans2) [inline storm-core/src/jvm/org/apache/storm/command/UploadCredentials.java]: My bad.  Please rename org.apache.storm.command.List to something else, like ListTopos, or whateve, and update the corresponding python and windows command files.

Comment (abhishekagarwal87) [inline storm-core/src/jvm/org/apache/storm/command/UploadCredentials.java]: Thanks @revans2 @unsleepy22 - Done

File: storm-core/src/jvm/org/apache/storm/command/UploadCredentials.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        if (null != rawCredentials) {
+            for (int i = 0; i < rawCredentials.size(); i += 2) {
+                credentialsMap.put(rawCredentials.get(i), rawCredentials.get(i + 1));
+            }
+        }
+        if (null != credentialFile) {
+            Properties credentialProps = new Properties();
+            credentialProps.load(new FileReader(credentialFile));
+            credentialsMap.putAll(credentialProps);
+        }
```
Comment (revans2) [inline storm-core/src/jvm/org/apache/storm/command/UploadCredentials.java]: In the original code merge gives precedence to later parameters.  So here rawCredentials should win over things in the properties file.

```
> (merge {:a 1} {:a 2})
{:a 2}
```

Comment (abhishekagarwal87) [inline storm-core/src/jvm/org/apache/storm/command/UploadCredentials.java]: Yes. Restructured the code to have the same behavior.

**Comment by revans2** [comment]
+1 pending travis

**Comment by revans2** [comment]
The test failure looks unrelated.  It looks like a race in one of the tests that we lost.

+1

**Pull request merged**